### PR TITLE
Add support for Audio Events

### PR DESCRIPTION
**Pull request by HennerM**

### CODE REVIEW

File: speechmatics/cli.py [hunk 1]
```diff
@@ -341,6 +342,10 @@ def get_transcription_config(
     if args_auto_chapters or auto_chapters_config is not None:
         config["auto_chapters_config"] = AutoChaptersConfig()
 
+    arg_audio_events = args.get("audio_events")
```
Comment (dumitrugutu) [inline speechmatics/cli.py:345]: Should also allow audio events through config file

File: speechmatics/cli_parser.py [hunk 2]
```diff
@@ -467,6 +467,10 @@ def get_arg_parser():
         help="Which type of diarization to use.",
     )
 
+    rt_transcribe_command_parser.add_argument(
+        "--audio-events", action="store_true", help="Print audio events to the console"
```
Comment (dumitrugutu) [inline speechmatics/cli_parser.py:471]: What constitutes an audio event?

File: speechmatics/client.py [hunk 3]
```diff
@@ -72,7 +72,6 @@ def __init__(
             self.connection_settings.set_missing_values_from_config(UsageMode.RealTime)
         self.websocket = None
         self.transcription_config = None
-        self.translation_config = None
```
Comment (dumitrugutu) [inline speechmatics/client.py:75]: Since this is introducing potentially breaking changes to existing functionality it should be done in a separate MR and released as minor client version? Otherwise anyone interested in using the new audio-events feature could potentially be impacted.

File: speechmatics/client.py [hunk 4]
```diff
@@ -435,7 +448,6 @@ async def run(
             consumer/producer tasks.
         """
         self.transcription_config = transcription_config
```
Comment (jrg1381) [inline speechmatics/client.py:450]: Would this be a better place for checking if `transcription_config` is `None`, like the added asserts in lines 137 and 164? Did you find some circumstances where it was `None`?

File: speechmatics/cli_parser.py (hunk 2 above)
Comment (jrg1381) [reply speechmatics/cli_parser.py:471]: This is a good point, you need a bit more exposition in the Python client help, so people don't have to resort to the main documentation.

**Comment by aaronng91** [comment]
add some unit tests to tests/test_cli and tests/test_models

File: speechmatics/client.py (hunk 4 above)
Comment (HennerM) [reply speechmatics/client.py:450]: The assertions is mostly to satisfy any type checkers, to know that this is actually always a value.

**Comment by HennerM** [comment]
> add some unit tests to tests/test_cli and tests/test_models

I don't think these tests are very useful, the test_cli are basically testing argparse and test_models test mostly the asdict method in `data classes`, or am I missing something?

File: speechmatics/client.py (hunk 3 above)
Comment (HennerM) [reply speechmatics/client.py:75]: Can I not just change the minor version on here? The audio-events feature will be on top of that bc breaking change regardless?

File: speechmatics/cli.py (hunk 1 above)
Comment (HennerM) [reply speechmatics/cli.py:346]: good catch, done

**Comment by aaronng91** [comment]
> > add some unit tests to tests/test_cli and tests/test_models
> 
> I don't think these tests are very useful, the test_cli are basically testing argparse and test_models test mostly the asdict method in `data classes`, or am I missing something?

yea its not a big deal in this case, but i think it can be useful to detect errors when you have many config options, and allos you to test the client without building it

**Review by aaronng91 — APPROVED** [review]

File: speechmatics/cli_parser.py [hunk 5]
```diff
@@ -467,6 +467,10 @@ def get_arg_parser():
         help="Which type of diarization to use.",
     )
 
+    rt_transcribe_command_parser.add_argument(
+        "--audio-events", action="store_true", help="Enable audio event detection and print events in square-brakcets to the console, e.g. [MUSIC]"
```
Comment (jrg1381) [inline speechmatics/cli_parser.py:471]:
```suggestion
        "--audio-events", action="store_true", help="Enable audio event detection and print events in square-brackets to the console, e.g. [MUSIC]"
```
